nixos/modules/services/security/certmgr.nix

```diff
@@ -0,0 +1,210 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.certmgr;
+
+  allSpecs = pkgs.buildEnv {
+    name = "certmgr.d";
+    paths = mapAttrsToList (name: value: pkgs.writeTextDir (name + ".json") (builtins.toJSON value)) cfg.declSpecs
+         ++ optional (cfg.impSpecs != []) (pkgs.linkFarm "impSpecs" cfg.impSpecs);
+  };
+
+  certmgrYaml = pkgs.writeText "certmgr.yaml" (builtins.toJSON {
+    dir = allSpecs;
+    default_remote = cfg.defaultRemote;
+    svcmgr = cfg.svcManager;
+    inherit (cfg) before interval metricsPort metricsAddress;
+  });
+
+  # Clean list by making sure:
+  # - all entries are at least 2 chars long (we don't want "/" or " " or ".")
+  # - all entries are unique
+  uniqueFilter = list: unique (filter (p: (stringLength p) > 2) list);
+
+  # Traverse all attributes matching predicate recursively and return the directory names of the attr-values as a list
+  catDirsRecursive = pred: set:
+    (concatLists (mapAttrsToList (n: v:
+        (if isAttrs v then (catDirsRecursive pred v)
+        else if (pred n v) then [(dirOf v)]
+        else [""])
+      ) set));
+
+  # If cfg.mkTargetDirs is true,
+  # .. select all attributes named "path" under declSpecs and resolve their directories for mkdir
+  preStart = concatStringsSep "\n" [
+    (optionalString cfg.mkTargetDirs
+      (concatStringsSep " "
+        (["mkdir -p"] ++ (uniqueFilter (catDirsRecursive (n: v: n == "path") cfg.declSpecs)))
```

Comment:
Changed as part of https://github.com/NixOS/nixpkgs/pull/44556/commits/e77155cbf3eafde44bc923e6f19b0e974a86c8f2. Using `filterAttrsRecursive`. Escaping shell-args. But still using nix for `dirOf`and `unique`, since I don't see the reason for doing this at runtime (in bash)?

Tell me what you think about the overall solution, please :) - after merge of impSpecs and declSpecs, that is.